src/view/com/post-thread/PostThread.tsx

```diff
@@ -365,20 +414,29 @@ function PostThreadLoaded({
       ref={ref}
       data={posts}
       initialNumToRender={!isNative ? posts.length : undefined}
-      maintainVisibleContentPosition={
-        !needsScrollAdjustment.current
-          ? MAINTAIN_VISIBLE_CONTENT_POSITION
-          : undefined
-      }
       keyExtractor={item => item._reactKey}
       renderItem={renderItem}
       refreshing={isPTRing}
       onRefresh={onPTR}
-      onContentSizeChange={onContentSizeChange}
+      // Always keep 2 pages rendered up top
+      onStartReached={onStartReached}
       style={s.hContentRegion}
+      // We only run this on web, since maintainVisibleContentPosition doesn't work there
+      onContentSizeChange={isWeb ? onContentSizeChange : undefined}
+      maintainVisibleContentPosition={
+        isWeb ? undefined : MAINTAIN_VISIBLE_CONTENT_POSITION
+      }
+      ListHeaderComponent={
+        isTabletOrMobile ? (
+          <ViewHeader
+            title={_(msg({message: `Post`, context: 'description'}))}
+          />
+        ) : undefined
+      }
```

Comment:
I'm moving the header render to the `ListHeaderComponent` so that we can simplify the logic in slicing the parents array. It also prevents some layout issues when we have to load more items when we scroll up. The behavior seems to be identical to what we currently have.

Reply:
Nevermind, it makes more sense to do this how we did before.